proto/api/v1/remote_runner.proto

```diff
@@ -61,12 +62,31 @@ message RunRequest {
 
   // Whether to skip the automatic GitHub setup steps on the remote runner.
   bool skip_auto_checkout = 12;
+
+  // DEPRECATED: Use wait_mode instead.
+  // If true, start the runner but do not wait for the action to be scheduled
+  // before returning a response.
+  bool async = 8 [deprecated = true];
 }
 
 message Step {
   string run = 1;
 }
 
+enum WaitCondition {
+  // Unknown wait condition.
+  UNKNOWN_CONDITION = 0;
+
+  // Wait until the invocation is created but not complete.
+  CREATED = 1;
+
+  // Wait for the invocation to complete.
+  COMPLETE = 2;
+  
+  // Return immediately without waiting.
+  IMMEDIATE = 3;  
```

Comment:
maybe CREATED => STARTED and IMMEDIATE => QUEUED would be more accurate and possibly more intuitive. Could also reorder the enum values so that they go in order of chronology (SCHEDULED, STARTED, COMPLETE)

Reply:
So good - done!